### PR TITLE
Decompiled ovl_Demo_Im

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: src/overlays/actors/ovl_Demo_Im/z_demo_im.c [hunk 1]
```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
...
+    } else {
+        if (*something == 0) {
+            *something = 1;
+        }
+    }
+}
+
+void func_80984D00(DemoIm* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider = &this->collider;
+    if (this) {} // needed to match
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Im/z_demo_im.c:155]: These can probably be replaced by not using a collider temp and by having this function take `Actor* thisx`, or something along those lines.

File: src/overlays/actors/ovl_Demo_Im/z_demo_im.c [hunk 2]
```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
...
+
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_SetCylinder_Set3(globalCtx, collider, &this->actor, &sCylinderInit);
+}
+
+void func_80984D4C(DemoIm* this, GlobalContext* globalCtx) {
+    ColliderCylinder* collider = &this->collider;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Im/z_demo_im.c:162]: Similar thing here, the collider temp is very unlikely to be a real thing, but this function could very well take `Actor* thisx`

File: src/overlays/actors/ovl_Demo_Im/z_demo_im.c [hunk 3]
```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
...
+void func_80984DB8(DemoIm* this) {
+    s32 pad[2];
+    Vec3s* vec1 = &this->unk_2D4.unk_08;
+    Vec3s* vec2 = &this->unk_2D4.unk_0E;
+
+    Math_SmoothScaleMaxMinS(&vec1->x, 0, 0x14, 0x1838, 0x64);
+    Math_SmoothScaleMaxMinS(&vec1->y, 0, 0x14, 0x1838, 0x64);
+
+    Math_SmoothScaleMaxMinS(&vec2->x, 0, 0x14, 0x1838, 0x64);
+    Math_SmoothScaleMaxMinS(&vec2->y, 0, 0x14, 0x1838, 0x64);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Im/z_demo_im.c:183]: All these values should probably be decimal

File: src/overlays/actors/ovl_Demo_Im/z_demo_im.c [hunk 4]
```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
...
+        if (*unk_278 >= 0.0f) {
+            if (this->unk_27C == 0) {
+                Vec3f* thisPos = &this->actor.posRot.pos;
+                s16 shapeRotY = this->actor.shape.rot.y;
+                f32 spawnPosX = thisPos->x + (Math_Sins(shapeRotY) * 30.0f);
+                f32 spawnPosY = thisPos->y;
+                f32 spawnPosZ = thisPos->z + (Math_Coss(shapeRotY) * 30.0f);
+
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ARROW, spawnPosX, spawnPosY, spawnPosZ, 0xFA0,
+                            this->actor.shape.rot.y, 0, ~0x9);
```
Comment (Roman971) [inline src/overlays/actors/ovl_Demo_Im/z_demo_im.c:748]:
```suggestion
                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ARROW, spawnPosX, spawnPosY, spawnPosZ, 0xFA0,
                            this->actor.shape.rot.y, 0, 0xFFF6);
```

File: src/overlays/actors/ovl_Demo_Im/z_demo_im.c [hunk 5]
```diff
@@ -21,229 +94,1125 @@ const ActorInit Demo_Im_InitVars = {
...
+    this->action = 7;
+    this->drawConfig = 0;
+    this->unk_26C = 0;
+    this->unk_270 = 0;
+    this->actor.shape.unk_14 = 0;
+    this->unk_268 = 0.0f;
+}
+
+void func_80984C8C(DemoIm* this, GlobalContext* globalCtx) {
+    u32* something = &D_8098783C;
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/overlays/actors/ovl_Demo_Im/z_demo_im.c:137]: From what I can tell, yes - it's the same case in demo_sa's func_8098E554

**Pull request merged**